test/run

```diff
@@ -0,0 +1,158 @@
+        mongo_admin_cmd "db=db.getSiblingDB('${DB}');db.testData.insert({x:0});"
+        mongo_cmd "db.createUser({user:'test_user2',pwd:'test_password2',roles:['readWrite']});"
+    fi
+    echo "  Testing user privileges"
+    mongo_cmd "db.testData.insert({ y : 1 });"
+    mongo_cmd "db.testData.insert({ z : 2 });"
+    mongo_cmd "db.testData.find().forEach(printjson);"
+    mongo_cmd "db.testData.count();"
+    mongo_cmd "db.testData.drop();"
+    mongo_cmd "db.dropDatabase();"
+    echo "  Success!"
+}
+
+function ctest_configuration() {
+    echo "  Testing image configuration settings"
+    test_config_option MONGODB_QUIET true "quiet: true"
+    if test $VERSION_NUM -le 30; then
+        test_config_option MONGODB_PREALLOC false "preallocDataFiles: false"
+        test_config_option MONGODB_NOPREALLOC true "preallocDataFiles: false"
+        test_config_option MONGODB_SMALLFILES true "smallFiles: true"
+    fi
+    echo "  Success!"
+}
+
+function ctest_mount_config() {
+    local name="mount_config"
+    echo "  Testing config file mount"
+    local database='db'
+    local user='user'
+    local password='password'
+    local admin_password='adminPassword'
+
+    local volume_dir
+    local config_content
+    volume_dir=`mktemp -d --tmpdir mongodb-testdata.XXXXX`
+    chmod a+rwx ${volume_dir}
+    config_file=$volume_dir/mongod.conf
+    config_content="dbpath=/var/lib/mongodb/dbpath
+unixSocketPrefix = /var/lib/mongodb"
+    if test $VERSION_NUM -le 30; then
+    config_content="$config_content
+smallfiles = true
+noprealloc = true"
+    fi
+    echo "$config_content" > $config_file
+    chmod a+wr ${config_file}
+
+    CONTAINER_ARGS="
+-e MONGODB_DATABASE=${database}
+-e MONGODB_USER=${user}
+-e MONGODB_PASSWORD=${password}
+-e MONGODB_ADMIN_PASSWORD=${admin_password}
+-v ${config_file}:/etc/mongod.conf:Z
+-v ${volume_dir}:/var/lib/mongodb/dbpath:Z
+"
+    ct_create_container $name
+
+    # need to set these because `mongo_cmd` relies on global variables
+    USER=${user}
+    PASS=${password}
+    DB=${database}
+
+    # need this to wait for the container to start up
+    CONTAINER_IP=$(ct_get_cip ${name})
+    echo "  Testing mongod is running"
+    test_connection ${name}
+    echo "  Testing config file works"
+    docker exec $(ct_get_cid ${name}) bash -c "test -S /var/lib/mongodb/mongodb-27017.sock"
+
+    # need to remove volume_dir with sudo because of permissions of files written
+    # by the Docker container
+    sudo rm -rf ${volume_dir}
+
+    echo "  Success!"
+}
+
+function ctest_s2i() {
+    echo "  Testing s2i usage"
+    s2i usage ${S2I_ARGS} ${IMAGE_NAME} &>/dev/null
+
+    # Set configuration file (differs for mmapv1 storage engine)
+    mv ${TEST_DIR}/test-image/mongodb-cfg/mongod.conf ${TEST_DIR}/test-image/mongodb-cfg/mongod.conf.backup
+
+    if test $VERSION_NUM -ge 32; then
```

Comment:
@pkubatrh Thanks for fixing this problem.

Tests reports: `test/run: line 137: test_dir: unbound variable`